core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -164,18 +167,22 @@ class OptimizeExecutor(
 
   def optimize(): Seq[Row] = {
     recordDeltaOperation(txn.deltaLog, "delta.optimize") {
-      val minFileSize = sparkSession.sessionState.conf.getConf(
-        DeltaSQLConf.DELTA_OPTIMIZE_MIN_FILE_SIZE)
       val maxFileSize = sparkSession.sessionState.conf.getConf(
         DeltaSQLConf.DELTA_OPTIMIZE_MAX_FILE_SIZE)
-      require(minFileSize > 0, "minFileSize must be > 0")
       require(maxFileSize > 0, "maxFileSize must be > 0")
-
+      val (minFileSize, maxDeletedRowsRatio) = if (isPurge) {
+        (0L, 0d) // Only selects files with DV
+      } else {
+        val minFileSize = sparkSession.sessionState.conf.getConf(
+          DeltaSQLConf.DELTA_OPTIMIZE_MIN_FILE_SIZE)
+        val maxDeletedRowsRatio = sparkSession.sessionState.conf.getConf(
+          DeltaSQLConf.DELTA_OPTIMIZE_MAX_DELETED_ROWS_RATIO)
+        require(minFileSize > 0, "minFileSize must be > 0")
+        (minFileSize, maxDeletedRowsRatio)
+      }
```

Comment:
Also does this select the files that don't have DV and gets rewritten? Do we want that?

Reply:
It didn't select files without DVs, because with `minFileSize` set to `0` all files are de-selected unless they contain DVs. Inside `pruneCandidateFileList()`:
```
    // Select files that are small or have too many deleted rows
    files.filter(
      addFile => addFile.size < minFileSize || shouldCompactBecauseOfDeletedRows(addFile))
```